testplan/common/utils/match.py

```diff
@@ -95,28 +101,42 @@ def match(self, regex, timeout=5):
         end of the file. If a match is found the line number is stored and the
         match is returned. If no match is found an Exception is raised.
 
-        :param regex: compiled regular expression (``re.compile``)
-        :type regex: ``re.Pattern``
+        :param regex: regex string or compiled regular expression
+            (``re.compile``)
+        :type regex: ``Union[str, re.Pattern]``
 
         :return: The regex match or raise an Exception if no match is found.
         :rtype: ``re.Match``
         """
         match = None
-        end_time = time.time() + timeout
+        start_time = time.time()
+        end_time = start_time + timeout
+
+        # As a convenience, we create the compiled regex if a string was
+        # passed.
+        if not hasattr(regex, 'match'):
+            regex = re.compile(regex)
+
         with open(self.log_path, 'r') as log:
             log.seek(self.position)
+
             while match is None:
                 line = log.readline()
                 if line:
                     match = regex.match(line)
                     if match:
-                        self.position = log.tell()
                         break
-                if time.time() > end_time:
-                    self.position = log.tell()
-                    break
-                time.sleep(LOG_MATCHER_INTERVAL)
+                else:
+                    time.sleep(LOG_MATCHER_INTERVAL)
```

Comment:
Use get_sleeper utility?

Reply:
The `get_sleeper()` utility is for sleeping after every iteration I think? Which is not what this change is doing: we only want to sleep when no lines are read (i.e. we are waiting on the application to write a line), not when lines are being read - so that we can iterate quickly through thousands of lines without unnecessary sleeping.